LogicalTypes.md

```diff
@@ -648,49 +650,105 @@ optional group my_list (LIST) {
 }
 ```
 
-Some existing data does not include the inner element layer. For
-backward-compatibility, the type of elements in `LIST`-annotated structures
-should always be determined by the following rules:
+##### 2-level structure
+
+Some existing data does not include the inner element layer, meaning that `LIST`
+annotates a 2-level structure. In contrast to 3-level structure, the repetition
+of 2-level structure can be `optional`, `required`, or `repeated`.
```

Comment:
I find this confusing. Why does it need to change?

Reply:
I just happen to see that this section is all about 2-level list so I have added a title to make it explicit.

> The problem is when we need to decide whether a structure is a 2-level or 3-level.

The change aims to exhaust all possible forms of list structures and we can just apply these rules in order to deduce a list type:
- regular 3-level list
- 3-level list (w/ different names)
- 2-level list
- 1-level list (w/o list annotation)

Reply:
I decide to keep the subtitles as they are pretty clear. It is the exact order to apply when deducing a list type. Please let me know if you have different opinions and I'm happy to change. @rdblue @gszadovszky @emkornfield